front/components/assistant/conversation/input_bar/InputBarAttachments.tsx

```diff
@@ -10,57 +10,88 @@ import {
   SearchInputWithPopover,
   Separator,
 } from "@dust-tt/sparkle";
+import type {
+  DataSourceViewContentNode,
+  LightWorkspaceType,
+} from "@dust-tt/types";
 import { MIN_SEARCH_QUERY_SIZE } from "@dust-tt/types";
 import { useEffect, useMemo, useRef, useState } from "react";
 
 import type { FileUploaderService } from "@app/hooks/useFileUploaderService";
+import {
+  getLocationForDataSourceViewContentNode,
+  getVisualForDataSourceViewContentNode,
+} from "@app/lib/content_nodes";
+import { useSpaces, useSpacesSearch } from "@app/lib/swr/spaces";
 
 interface InputBarAttachmentsProps {
+  owner: LightWorkspaceType;
   fileUploaderService: FileUploaderService;
-  onConnectedFileSelect: (fileId: string) => void;
+  onNodeSelect: (node: DataSourceViewContentNode) => void;
   isLoading?: boolean;
 }
 
-// TODO(attach from input bar): use component with spaces wide search
 export const InputBarAttachments = ({
+  owner,
   fileUploaderService,
-  onConnectedFileSelect,
+  onNodeSelect,
   isLoading = false,
 }: InputBarAttachmentsProps) => {
   const fileInputRef = useRef<HTMLInputElement>(null);
   const [search, setSearch] = useState("");
   const [debouncedSearch, setDebouncedSearch] = useState<string>("");
+  const [isDebouncing, setIsDebouncing] = useState(false);
 
-  const items = Array.from({ length: 50 }).map((_, i) => ({
-    id: `${i}`,
-    title: `Document ${i + 1}`,
-    path: "Github",
-  }));
-
-  const filteredItems = useMemo(() => {
-    return items.filter((item) => item.title.includes(debouncedSearch));
-  }, [debouncedSearch, items]);
+  const { spaces, isSpacesLoading } = useSpaces({ workspaceId: owner.sId });
+  const { searchResultNodes, isSearchLoading } = useSpacesSearch({
+    includeDataSources: true,
+    owner,
+    search: debouncedSearch,
+    viewType: "all",
+    disabled: isSpacesLoading,
+    spaceIds: spaces.map((s) => s.sId),
+  });
 
   useEffect(() => {
+    setIsDebouncing(true);
     const timeout = setTimeout(() => {
       setDebouncedSearch(search.length >= MIN_SEARCH_QUERY_SIZE ? search : "");
+      setIsDebouncing(false);
     }, 300);
     return () => {
       clearTimeout(timeout);
+      setIsDebouncing(false);
     };
   }, [search]);
 
+  const spacesMap = useMemo(
+    () => Object.fromEntries(spaces.map((space) => [space.sId, space.name])),
+    [spaces]
+  );
+
+  const unfoldedNodes: DataSourceViewContentNode[] = useMemo(
+    () =>
+      searchResultNodes.flatMap((node) => {
+        const { dataSourceViews, ...rest } = node;
+        return dataSourceViews.map((view) => ({
+          ...rest,
+          dataSourceView: view,
```

Comment:
```suggestion
          dataSourceViews: [view],
```
to comply with the type specified and not create a new one (and you can remove the exploding above)